agent/agent.go

```diff
@@ -165,73 +155,86 @@ func formatBytes(val uint64) string {
 	return fmt.Sprintf("%d bytes", val)
 }
 
-func handle(conn io.Writer, msg []byte) error {
-	switch msg[0] {
+// Agent implement http.Handler
+type Agent struct{}
+
+// ServeHTTP parse the requests body into signal.Command and run the task
+func (a *Agent) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	cmd := signal.Command{}
+	decodeErr := json.NewDecoder(r.Body).Decode(&cmd)
```

Comment:
I keep the behavior of "agent.Listen" function (for starting server at random port or the one specific with Options.Addr) but also keep this Agent public for advance usage.
For example most of my company's services already a HTTP server, the company convention restricted service on certain port. So with this public Agent handler I can easily add this to the current HTTP mux.